tests/regression/nextcloud-rule-exclusions-plugin/9508880.yaml

```diff
@@ -20,84 +20,10 @@ tests:
-            dest_addr: 127.0.0.1
-            headers:
-              Host: localhost
-              User-Agent: OWASP ModSecurity Core Rule Set
-              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-            port: 80
-            method: POST
-            uri: /ocs/v2.php/references/resolve
-            data: "reference=https://www.jaycar.com.au/raspberry-pi-3-switchmode-power-supply-5-1v-2-5a-with-usb-micro-b/p/MP3536?pos=1&queryId=0a9ebf34f466c42c586d9561f4ac20d2&sort=relevance&searchText=2.5a%205v%20micro%20usb"
-          output:
-            no_log_contains: id "932200"
-  - test_title: 9508880-4
-    desc: Adding a hyperlink to text
-    stages:
-      - stage:
-          input:
-            dest_addr: 127.0.0.1
-            headers:
-              Host: localhost
-              User-Agent: OWASP ModSecurity Core Rule Set
-              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-            port: 80
-            method: POST
-            uri: /ocs/v2.php/references/resolve
-            data: "reference=https://www.jaycar.com.au/raspberry-pi-3-switchmode-power-supply-5-1v-2-5a-with-usb-micro-b/p/MP3536?pos=1&queryId=0a9ebf34f466c42c586d9561f4ac20d2&sort=relevance&searchText=2.5a%205v%20micro%20usb"
-          output:
-            no_log_contains: id "942430"
-  - test_title: 9508880-5
-    desc: Adding a hyperlink to text
-    stages:
-      - stage:
-          input:
-            dest_addr: 127.0.0.1
-            headers:
-              Host: localhost
-              User-Agent: OWASP ModSecurity Core Rule Set
-              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-            port: 80
-            method: POST
-            uri: /ocs/v2.php/references/resolve
-            data: "reference=https://www.jaycar.com.au/raspberry-pi-3-switchmode-power-supply-5-1v-2-5a-with-usb-micro-b/p/MP3536?pos=1&queryId=0a9ebf34f466c42c586d9561f4ac20d2&sort=relevance&searchText=2.5a%205v%20micro%20usb"
-          output:
-            no_log_contains: id "942431"
-  - test_title: 9508880-6
-    desc: Adding a hyperlink to text
-    stages:
-      - stage:
-          input:
-            dest_addr: 127.0.0.1
-            headers:
-              Host: localhost
-              User-Agent: OWASP ModSecurity Core Rule Set
-              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-            port: 80
-            method: POST
-            uri: /ocs/v2.php/references/resolve
-            data: "reference=https://www.jaycar.com.au/raspberry-pi-3-switchmode-power-supply-5-1v-2-5a-with-usb-micro-b/p/MP3536?pos=1&queryId=0a9ebf34f466c42c586d9561f4ac20d2&sort=relevance&searchText=2.5a%205v%20micro%20usb"
-          output:
-            no_log_contains: id "942432"
+            no_log_contains: |
```

Comment:
Careful! This is a regular expression, not a list.

Reply:
so what should I do here then?
like I mentioned in your other review, I didn't want to create multiple tests, testing the same payload just to test different rules.

Reply:
For example:
```yaml
            no_log_contains: |
              id 942(?:432|...)
```

Reply:
gotcha, but that looks a bit worse in terms of readability, is this a performance change?

Reply:
Maybe not as readable, but, as you wrote about the tests you removed, more readable than duplicated tests with different `no_log_contains:`.

> is this a performance change?

I don't understand the question. Are you asking whether it's a new thing that `no_log_contains` is a regular expression?

Reply:
no, I'm just trying to understand the reasoning for change

Reply:
Not sure, what you are referring to. `log_contains` and `no_log_contains` have been regular expressions for a long time (maybe always, not sure).

Reply:
I just mean why I should use a non capturing group to reduce the amount of rule ID entries like in your example?
I think it's perfectly fine to list all of the rule IDs as is

Reply:
Oh, I see :D. Yes, feel free to drop the group, that was just an example.

Reply:
so I can leave them as is then as just a list?

Reply:
No, you have to at least transform the list into `id (?:942432|942...)`.  What you currently have there would amount to `id 942432\nid 942...`. So you'd be looking for the _entire string_.

Reply:
ok... everything should be good then

Reply:
ooh, you just want all of the rule IDs to be on the same line, you don't care if I use something like non-capturing groups or character sets, or do you want both?

Reply:
You can write the regex however you like :)

Reply:
is this better?

Reply:
Almost, but now you have extra spaces that are part of your match expression.